### DISCUSSION TITLE
Invalid token: token contains an invalid number of segments

### DISCUSSION THREAD

**Discussion by edcelmanuel** [supabase/supabase#1757] [Questions] [answered]
Im getting error when i try to change password with a recovery access token got from my email. 

how can i fix this

```js
const { error, data } = await supabase.auth.api.updateUser("qnINtA9aNubFAkE9qZOiUA", { password: "password" })
```

**Comment by inian** [comment, marked as answer]
Hi @edcelmanuel, the first argument should be a valid logged in JWT. [Docs](https://supabase.io/docs/gotrue/client/api-updateuser)

**Reply by edcelmanuel** [reply]
Ohhh i see.

**Comment by nkabour** [comment]
Hi - i’m facing the same problem 
I’m working on password recovery feature however the resetPasswordForEmail doesn’t embed all the query params it will only embed the token, type of request and redirectTo in the link 

and when i use the token in the url to update the password i get the same error 

i found a solution for that by calling /verify endpoint which will return a valid token - i wonder if that will be done by auth.api instead of calling goTrue endpoints 

btw, my application is self hosted so this might be the problem i’m not sure - please correct me if im wrong
